core/unify.ml

```diff
@@ -95,7 +101,15 @@ let rec eq_types : (datatype * datatype) -> bool =
          end
       | `Application (s, ts) ->
           begin match unalias t2 with
-              `Application (s', ts') -> s = s' && List.for_all2 (Utility.curry eq_type_args) ts ts'
+              `Application (s', ts') ->
+                s = s' && List.for_all2 (Utility.curry eq_type_args) ts ts'
+            | _ -> false
+          end
+      | `RecursiveApplication a1 ->
+          begin match unalias t2 with
```

Comment:
Redundant `begin ... end`?

Reply:
This one isn't redundant, since the next expression is another case branch.